### ISSUE TITLE
iCloud Password Extension

### ISSUE THREAD

**Issue by dbrewood** [ungoogled-software/ungoogled-chromium-macos#144]
Install the extension and try to sign it into iCloud via authentication. You are informed you must be using macOS Sonoma and are asked to download it. 
I'm running macOS Sonoma :)

**Comment by Cubik65536** [comment]
I can duplicate this issue, will investigate into it.

**Comment by Cubik65536** [comment]
Can confirm that this is also an issue with official Google Chromium, I do recommend you to report this to Apple and Google.

**Comment by dbrewood** [comment]
I don't have this issue with Brave or Vivaldi.....

**Comment by claudiodekker** [comment]
This has to do with that a `NativeMessagingHosts` file isn't being created for Chromium, but does for those other browsers.
To fix this you can copy the `com.apple.passwordmanager.json` from Vivaldi/Brave's `NativeMessagingHosts` folder, or simply run the following command in Terminal:

```bash
echo '{
    "name": "com.apple.passwordmanager",
    "description": "PasswordManagerBrowserExtensionHelper",
    "path": "/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper",
    "type": "stdio",
    "allowed_origins": [
        "chrome-extension://pejdijmoenmkgeppbflobdenhhabjlaj/",
        "chrome-extension://mfbcdcnpokpoajjciilocoachedjkima/"
    ]
}' > ~/Library/Application\ Support/Chromium/NativeMessagingHosts/com.apple.passwordmanager.json
```

**Comment by dbrewood** [comment]
Thanks that has indeed fixed the issue. Surely though there should not be a need to fix this?

**Comment by Cubik65536** [comment]
> Thanks that has indeed fixed the issue. Surely though there should not be a need to fix this?

This is an issue of vanilla Chromium (you may already discovered), as ungoogled-chromium wants to stay as close as possible to vanilla, we might not fix it here. I do recommend you to report this to Chromium dev directly though.

**Comment by PF4Public** [comment]
Well, let's consider it solved for now then.

**Comment by dbrewood** [comment]
To clarify, running the script above did indeed fix the problem, although I still think a permanent fix needs to be incorporated into the program itself. The need for manual fixes like this having to be applied is why I stopped using it.

**Comment by Cubik65536** [comment]
> To clarify, running the script above did indeed fix the problem, although I still think a permanent fix needs to be incorporated into the program itself. The need for manual fixes like this having to be applied is why I stopped using it.

`NativeMessagingHosts` should be installed by the **extension** rather than the browser, there’s absolutely nothing we can do, neither actually Chromium devs, you should report this to Apple if what you want is a permanent fix, otherwise I could only suggest you to use the temporary fix or stop using browsers that Apple haven’t considered at the first place.

**Comment by Cubik65536** [comment]
Reference: https://github.com/keeweb/keeweb/commit/e674607fce5e5c849b1396b5114325a9f6416820 (another password manager that uses NativeMessagingHosts)

**Comment by dbrewood** [comment]
Interesting, I can't see that Apple have considered Vivaldi or Brave and they both work.

**Comment by Cubik65536** [comment]
> Interesting, I can't see that Apple have considered Vivaldi or Brave and they both work.

Because Brave points their location to Chrome ([code](https://github.com/brave/brave-core/blob/8a22e0e27e19c7c89dd201abdcb3908566549ed9/app/brave_main_delegate.cc#L123-L127)). Which:

1. Chromium doesn't, obviously.
2. They actually shouldn't do that.

I could assume that Vivaldi do the same.

And I will go strongly against adding this for UGC.

**Comment by dbrewood** [comment]
Fair enough I understand your position.

**Comment by xcrap** [comment]
This is happening to be, i noticed it happened after installing several browsers to try, I already had arc but installed Vivaldi and Chrome Canary, after removing and going back to this extension this happened and the terminal command makes nothing, any ideias ?

**Comment by chefnaphtha** [comment]
the workaround stopped working in the latest macos update

**Comment by magn2o** [comment]
As of macOS 15.4, it appears that Apple introduced strict checking of `signing-identifier` and `team-identifier` in `/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper` (which you can verify through hex editing). There are a list of specific whitelisted identifiers, which, unfortunately ungoogled-chromium is not part of.

As far as I can tell, the list of whitelisted browsers are:

| signing-identifier | team-identifier |
| --- | --- |
| com.google.Chrome | EQHXZ8M8AV0U |
| com.google.Chrome.beta | EQHXZ8M8AV0W |
| com.google.Chrome.canary | EQHXZ8M8AV0T |
| com.microsoft.edgemac | UBF8T346G90Y |
| com.microsoft.edgemac.Beta | UBF8T346G90X |
| com.microsoft.edgemac.Dev | UBF8T346G90 |
| com.microsoft.edgemac.Canary | UBF8T346G90R |
| org.mozilla.firefox | 43AQ936H960b |
| org.mozilla.firefoxdeveloperedition | 43AQ936H960R |
| org.mozilla.nightly | 43AQ936H960Y |
| company.thebrowser.Browser | S6N382Y83G0P |
| com.brave.Browser | KL8N8XSYF40R |
| com.vivaldi.Vivaldi | 4XF3XNRN6Y0V |
| com.operasoftware.Opera | A2P9LX4JPN0X |
| com.operasoftware.OperaGX | A2P9LX4JPN0 |

**Comment by Cubik65536** [comment]
> As of macOS 15.4, it appears that Apple introduced strict checking of `signing-identifier` and `team-identifier` in `/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper` (which you can verify through hex editing). There are a list of specific whitelisted identifiers, which, unfortunately ungoogled-chromium is not part of.
> 
> As far as I can tell, the list of whitelisted browsers are:
> signing-identifier 	team-identifier
> com.google.Chrome 	EQHXZ8M8AV0U
> com.google.Chrome.beta 	EQHXZ8M8AV0W
> com.google.Chrome.canary 	EQHXZ8M8AV0T
> com.microsoft.edgemac 	UBF8T346G90Y
> com.microsoft.edgemac.Beta 	UBF8T346G90X
> com.microsoft.edgemac.Dev 	UBF8T346G90
> com.microsoft.edgemac.Canary 	UBF8T346G90R
> org.mozilla.firefox 	43AQ936H960b
> org.mozilla.firefoxdeveloperedition 	43AQ936H960R
> org.mozilla.nightly 	43AQ936H960Y
> company.thebrowser.Browser 	S6N382Y83G0P
> com.brave.Browser 	KL8N8XSYF40R
> com.vivaldi.Vivaldi 	4XF3XNRN6Y0V
> com.operasoftware.Opera 	A2P9LX4JPN0X
> com.operasoftware.OperaGX 	A2P9LX4JPN0

Unfortunately, it will probably impossible for us to get on that white list since ungoogled-chromium is literally signed with my *personal* account, and the chance of Apple trusting "me" is... let's say 0 :(

**Comment by chefnaphtha** [comment]
sad. if anyone's looking for an alternative i switched to bitwarden (vaultwarden) and while it's not as pretty and integrated it works fine

**Comment by Cubik65536** [comment]
Yeah it is sad, and other browsers (like Zen) also has this problem, we all hope that we'll come to a solution at some point.

**Comment by pedromarvarez** [comment]
> As of macOS 15.4, it appears that Apple introduced strict checking of `signing-identifier` and `team-identifier` in `/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper` (which you can verify through hex editing). There are a list of specific whitelisted identifiers, which, unfortunately ungoogled-chromium is not part of.
> 
> As far as I can tell, the list of whitelisted browsers are:
> 
> signing-identifier	team-identifier
> com.google.Chrome	EQHXZ8M8AV0U
> com.google.Chrome.beta	EQHXZ8M8AV0W
> com.google.Chrome.canary	EQHXZ8M8AV0T
> com.microsoft.edgemac	UBF8T346G90Y
> com.microsoft.edgemac.Beta	UBF8T346G90X
> com.microsoft.edgemac.Dev	UBF8T346G90
> com.microsoft.edgemac.Canary	UBF8T346G90R
> org.mozilla.firefox	43AQ936H960b
> org.mozilla.firefoxdeveloperedition	43AQ936H960R
> org.mozilla.nightly	43AQ936H960Y
> company.thebrowser.Browser	S6N382Y83G0P
> com.brave.Browser	KL8N8XSYF40R
> com.vivaldi.Vivaldi	4XF3XNRN6Y0V
> com.operasoftware.Opera	A2P9LX4JPN0X
> com.operasoftware.OperaGX	A2P9LX4JPN0

If I have Arc and i'm having the same problem, then why that is? It's happening since two months ago :( Is something on my end then?

**Comment by Cubik65536** [comment]
> > As of macOS 15.4, it appears that Apple introduced strict checking of `signing-identifier` and `team-identifier` in `/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper` (which you can verify through hex editing). There are a list of specific whitelisted identifiers, which, unfortunately ungoogled-chromium is not part of.
> > As far as I can tell, the list of whitelisted browsers are:
> > signing-identifier	team-identifier
> > com.google.Chrome	EQHXZ8M8AV0U
> > com.google.Chrome.beta	EQHXZ8M8AV0W
> > com.google.Chrome.canary	EQHXZ8M8AV0T
> > com.microsoft.edgemac	UBF8T346G90Y
> > com.microsoft.edgemac.Beta	UBF8T346G90X
> > com.microsoft.edgemac.Dev	UBF8T346G90
> > com.microsoft.edgemac.Canary	UBF8T346G90R
> > org.mozilla.firefox	43AQ936H960b
> > org.mozilla.firefoxdeveloperedition	43AQ936H960R
> > org.mozilla.nightly	43AQ936H960Y
> > company.thebrowser.Browser	S6N382Y83G0P
> > com.brave.Browser	KL8N8XSYF40R
> > com.vivaldi.Vivaldi	4XF3XNRN6Y0V
> > com.operasoftware.Opera	A2P9LX4JPN0X
> > com.operasoftware.OperaGX	A2P9LX4JPN0
> 
> If I have Arc and i'm having the same problem, then why that is? It's happening since two months ago :( Is something on my end then?

Well, `company.thebrowser.Browser` is Arc, so I can't tell if it is your issue or the list is not accurate...

**Comment by magn2o** [comment]
> > As of macOS 15.4, it appears that Apple introduced strict checking of `signing-identifier` and `team-identifier` in `/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper` (which you can verify through hex editing). There are a list of specific whitelisted identifiers, which, unfortunately ungoogled-chromium is not part of.
> > As far as I can tell, the list of whitelisted browsers are:
> > signing-identifier	team-identifier
> > com.google.Chrome	EQHXZ8M8AV0U
> > com.google.Chrome.beta	EQHXZ8M8AV0W
> > com.google.Chrome.canary	EQHXZ8M8AV0T
> > com.microsoft.edgemac	UBF8T346G90Y
> > com.microsoft.edgemac.Beta	UBF8T346G90X
> > com.microsoft.edgemac.Dev	UBF8T346G90
> > com.microsoft.edgemac.Canary	UBF8T346G90R
> > org.mozilla.firefox	43AQ936H960b
> > org.mozilla.firefoxdeveloperedition	43AQ936H960R
> > org.mozilla.nightly	43AQ936H960Y
> > company.thebrowser.Browser	S6N382Y83G0P
> > com.brave.Browser	KL8N8XSYF40R
> > com.vivaldi.Vivaldi	4XF3XNRN6Y0V
> > com.operasoftware.Opera	A2P9LX4JPN0X
> > com.operasoftware.OperaGX	A2P9LX4JPN0
> 
> If I have Arc and i'm having the same problem, then why that is? It's happening since two months ago :( Is something on my end then?

Confirm the bundle and team identifier of your installed app with `codesign -dv`.

Also, this particular issue would’ve appeared with macOS 15.4, so unless you were running betas, it should’ve only appeared a couple weeks ago.

**Comment by pedromarvarez** [comment]
Thank you for replying! @magn2o @Cubik65536 I already checked the bundle and team identifier and this is what i got from terminal:

**Identifier=company.thebrowser.Browser**
**TeamIdentifier=S6N382Y83G**

```
Executable=/Applications/Arc.app/Contents/MacOS/Arc
Identifier=company.thebrowser.Browser
Format=app bundle with Mach-O universal (x86_64 arm64)
CodeDirectory v=20500 size=597158 flags=0x10000(runtime) hashes=18650+7 location=embedded
VersionPlatform=1
VersionMin=851968
VersionSDK=983040
Hash type=sha256 size=32
CandidateCDHash sha256=0a84fcc6b89a95de3b2eda8beefcec3aafbc301b
CandidateCDHashFull sha256=0a84fcc6b89a95de3b2eda8beefcec3aafbc301b12a7e0fbde2b23f93a677ebe
Hash choices=sha256
CMSDigest=0a84fcc6b89a95de3b2eda8beefcec3aafbc301b12a7e0fbde2b23f93a677ebe
CMSDigestType=2
Executable Segment base=0
Executable Segment limit=65994752
Executable Segment flags=0x1
Page size=4096
CDHash=0a84fcc6b89a95de3b2eda8beefcec3aafbc301b
Signature size=9023
Authority=Developer ID Application: The Browser Company of New York Inc. (S6N382Y83G)
Authority=Developer ID Certification Authority
Authority=Apple Root CA
Timestamp=Apr 16, 2025 at 10:37:12
Info.plist entries=56
TeamIdentifier=S6N382Y83G
Runtime Version=15.0.0
Sealed Resources version=2 rules=13 files=836
Internal requirements count=1 size=220
pedromarvarez@Pedros-Mac-mini ~ % 
```

As well i have Sonoma 14.7.4 but i updated to Sequoia two months ago, i had to downgrade right away due a compatibility problem with my canon camera and since then i have this problem even though i'm in Sonoma 14.7.4.

I checked the passwords options and the Autofill and iCloud keychain is on, as well I toggled off and on just in case.
![Image](https://github.com/user-attachments/assets/107bcf0a-fd26-40a1-9b51-28325d494f79)

But i'm still having the same message when I try to use the extension in my Arc:
![Image](https://github.com/user-attachments/assets/7cc3621b-2612-4c12-87fd-10e5c78d8ea5)

**Comment by magn2o** [comment]
> As well i have Sonoma 14.7.4 but i updated to Sequoia two months ago, i had to downgrade right away due a compatibility problem with my canon camera and since then i have this problem even though i'm in Sonoma 14.7.4.

Wait, you're on 14.7? That being the case, this particular restriction does not apply to you. My guess is [this will address your issue](https://github.com/ungoogled-software/ungoogled-chromium-macos/issues/144#issuecomment-1913176798).

**Comment by Cubik65536** [comment]
It turns out that [disabling System Integrity Protection](https://developer.apple.com/documentation/security/disabling-and-enabling-system-integrity-protection) may be a temporary workaround. I just updated my macOS and since my SIP is already disabled for some other reason, my iCloud Password still works. I know this is absolutely not a good solution for most people, but it seems to be the only solution we have right now.

**Comment by dumbmoron** [comment]
I dug into this and was told that Apple will use data from [this repo](https://github.com/apple/password-manager-resources) to allowlist browsers:
> As of macOS Sequoia version 15.5 and above, information in this file is periodically ingested and re-packaged by Apple to limit the [Native Messaging Host](https://developer.chrome.com/docs/extensions/develop/concepts/native-messaging) of the iCloud Passwords extension to only communicate with known web browsers. 

I already submitted a PR to add UGC and it has been merged: https://github.com/apple/password-manager-resources/commit/5e310be024a0df4375733fde19690ff82117545a, so hopefully the extension will start working again in some future macOS release

**Comment by networkException** [comment]
Oh! That sounds great, thanks for looking into that!

**Comment by magn2o** [comment]
> I dug into this and was told that Apple will use data from [this repo](https://github.com/apple/password-manager-resources) to allowlist browsers:
> 
> > As of macOS Sequoia version 15.5 and above, information in this file is periodically ingested and re-packaged by Apple to limit the [Native Messaging Host](https://developer.chrome.com/docs/extensions/develop/concepts/native-messaging) of the iCloud Passwords extension to only communicate with known web browsers.
> 
> I already submitted a PR to add UGC and it has been merged: [apple/password-manager-resources@5e310be](https://github.com/apple/password-manager-resources/commit/5e310be024a0df4375733fde19690ff82117545a), so hopefully the extension will start working again in some future macOS release

This is great news(!), but it's worth noting this will only address the issue for ["official" builds](https://formulae.brew.sh/cask/eloston-chromium) as far as I can tell-- and thus x86_64 only. Until we can get arm64 in the official pipeline anyone using arm64 builds will still be unable to use Passwords integration, unfortunately. 😢

**Comment by magn2o** [comment]
> > This is great news(!), but it's worth noting this will only address the issue for ["official" builds](https://formulae.brew.sh/cask/eloston-chromium) as far as I can tell-- and thus x86_64 (Intel builds) only. Until we can get arm64 in the official pipeline anyone using arm64 builds will still be unable to use Passwords integration, unfortunately. 😢
> 
> [@magn2o](https://github.com/magn2o) We have both arm64 and x86 builds here on GitHub: https://github.com/ungoogled-software/ungoogled-chromium-macos/releases and they are signed with the same identifier / team ID, so they should both work. Honestly, I have no idea why the cask only has x86 builds.

Well I happily stand (sit?) corrected 🤣 This is great news all around! 🎉

**Comment by Cubik65536** [comment]
> I dug into this and was told that Apple will use data from [this repo](https://github.com/apple/password-manager-resources?rgh-link-date=2025-05-02T14%3A08%3A16.000Z) to allowlist browsers:
> 
> > As of macOS Sequoia version 15.5 and above, information in this file is periodically ingested and re-packaged by Apple to limit the [Native Messaging Host](https://developer.chrome.com/docs/extensions/develop/concepts/native-messaging) of the iCloud Passwords extension to only communicate with known web browsers.
> 
> I already submitted a PR to add UGC and it has been merged: [apple/password-manager-resources@5e310be](https://github.com/apple/password-manager-resources/commit/5e310be024a0df4375733fde19690ff82117545a), so hopefully the extension will start working again in some future macOS release

Wow! Thanks for looking into this issue! This sound great!

**Comment by t0bst4r** [comment]
> I dug into this and was told that Apple will use data from [this repo](https://github.com/apple/password-manager-resources) to allowlist browsers:
> 
> > As of macOS Sequoia version 15.5 and above, information in this file is periodically ingested and re-packaged by Apple to limit the [Native Messaging Host](https://developer.chrome.com/docs/extensions/develop/concepts/native-messaging) of the iCloud Passwords extension to only communicate with known web browsers.
> 
> I already submitted a PR to add UGC and it has been merged: [apple/password-manager-resources@5e310be](https://github.com/apple/password-manager-resources/commit/5e310be024a0df4375733fde19690ff82117545a), so hopefully the extension will start working again in some future macOS release

Awesome! Thank you!

Does it really require a macOS update? Is it known that this will be included in their next release or do they follow an additional (unknown) internal release cycle?

**Comment by 0xdevalias** [comment]
> As of macOS 15.4, it appears that Apple introduced strict checking of `signing-identifier` and `team-identifier` in `/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper` (which you can verify through hex editing).

@magn2o Curious if you're able to give a few more specific steps of what you were doing to inspect which identifiers are supported by a specific `PasswordManagerBrowserExtensionHelper` binary. Were you just looking at the output of `strings` or similar, or something else?

**Comment by 0xdevalias** [comment]
@hienlh Did you read these 2 comments that talk about why it changed in 15.4 and the solution to it?

- https://github.com/ungoogled-software/ungoogled-chromium-macos/issues/144#issuecomment-2781055694
- https://github.com/ungoogled-software/ungoogled-chromium-macos/issues/144#issuecomment-2847297391

**Comment by hienlh** [comment]
> [@hienlh](https://github.com/hienlh) Did you read these 2 comments that explain why it changed in 15.4 and the solution to it?
> 
> * [iCloud Password Extension #144 (comment)](https://github.com/ungoogled-software/ungoogled-chromium-macos/issues/144#issuecomment-2781055694)
> * [iCloud Password Extension #144 (comment)](https://github.com/ungoogled-software/ungoogled-chromium-macos/issues/144#issuecomment-2847297391)

Yes, read it. Now for other browsers (not ungoogled-chromium) we have to create a PR to this repo, right?
https://github.com/apple/password-manager-resources

**Comment by 0xdevalias** [comment]
> Now for other browsers (not ungoogled-chromium) we have to create a PR to this repo, right? [apple/password-manager-resources](https://github.com/apple/password-manager-resources?rgh-link-date=2025-06-04T01%3A17%3A41.000Z)

@hienlh That was my understanding based on reading this section of the README:

- https://github.com/apple/password-manager-resources#web-browser-extension-distribution-information

Though it would also be worth first checking if the browser you're wanting to add is already listed here (and that just hasn't been re-ingested into the version of macOS you're using):

- https://github.com/apple/password-manager-resources/blob/main/quirks/web-browser-extension-distribution-information.json

And probably also check that there aren't any existing issues/PRs for it:

- https://github.com/apple/password-manager-resources/issues
- https://github.com/apple/password-manager-resources/pulls

**Comment by CaptainWynn** [comment]
> > I dug into this and was told that Apple will use data from [this repo](https://github.com/apple/password-manager-resources) to allowlist browsers:
> > > As of macOS Sequoia version 15.5 and above, information in this file is periodically ingested and re-packaged by Apple to limit the [Native Messaging Host](https://developer.chrome.com/docs/extensions/develop/concepts/native-messaging) of the iCloud Passwords extension to only communicate with known web browsers.
> > 
> > 
> > I already submitted a PR to add UGC and it has been merged: [apple/password-manager-resources@5e310be](https://github.com/apple/password-manager-resources/commit/5e310be024a0df4375733fde19690ff82117545a), so hopefully the extension will start working again in some future macOS release
> 
> included in macOS 26 beta:
> 
> ![Image](https://github.com/user-attachments/assets/b8af7397-4017-4a4b-a4d8-e682aee627ab)

this is a great news! as far as i can see, zen is there in macos 26. anyone can confirm it works?

**Comment by fredngo** [comment]
@arvin-z 

Hum, I upgraded to macOS 15.6 and am running Ungoogled Chromium 138.0.7204.183 (the latest as far as I can tell).

What am I doing wrong?

<img width="459" height="234" alt="Image" src="https://github.com/user-attachments/assets/12b4788f-6b18-4b75-a2a1-b94768e96f24" />

<img width="490" height="138" alt="Image" src="https://github.com/user-attachments/assets/5b2e96bf-8d1c-4542-8d7d-67e897855c5e" />

**Comment by Qubik65536** [comment]
@fredngo Please check your `~/Library/Application\ Support/Chromium/NativeMessagingHosts/com.apple.passwordmanager.json`. I think the following steps (from @claudiodekker at the beginning of this thread) may still be needed:

> This has to do with that a `NativeMessagingHosts` file isn't being created for Chromium, but does for those other browsers.
> 
> To fix this you can copy the `com.apple.passwordmanager.json` from Vivaldi/Brave's `NativeMessagingHosts` folder into Chromium's, or simply run the following command in Terminal:
> 
> echo '{
>     "name": "com.apple.passwordmanager",
>     "description": "PasswordManagerBrowserExtensionHelper",
>     "path": "/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper",
>     "type": "stdio",
>     "allowed_origins": [
>         "chrome-extension://pejdijmoenmkgeppbflobdenhhabjlaj/",
>         "chrome-extension://mfbcdcnpokpoajjciilocoachedjkima/"
>     ]
> }' > ~/Library/Application\ Support/Chromium/NativeMessagingHosts/com.apple.passwordmanager.json

**Comment by fredngo** [comment]
@Qubik65536 Yup, that did it! The `com.apple.passwordmanager.json` file was missing. Working now! 🎉

Will this file be auto-created in the future somehow? Not sure we can close this issue before that happens...

**Comment by Qubik65536** [comment]
> [@Qubik65536](https://github.com/Qubik65536) Yup, that did it! The `com.apple.passwordmanager.json` file was missing. Working now! 🎉

Good news!

> FWIW, I didn't do those steps because of [this report](https://github.com/ungoogled-software/ungoogled-chromium-macos/issues/144#issuecomment-2936218027) that it no longer works as of macOS 15.4...

It didn't work as Chromium was not added to the known browser list as of 15.4, but @dumbmoron did it (see his comments above), and it is what made that step work *again*.

> Will this file be auto-created in the future somehow? Not sure we can close this issue before that happens...

This is something to be done by Apple not us, so unfortunately this is no longer a UGC issue, and I don't think Apple is going to do this nor us having any change to request them to do so anytime soon.

**Comment by robbymanis** [comment]
> [@fredngo](https://github.com/fredngo) Please check your `~/Library/Application\ Support/Chromium/NativeMessagingHosts/com.apple.passwordmanager.json`. I think the following steps (from [@claudiodekker](https://github.com/claudiodekker) at the beginning of this thread) may still be needed:
> 
> > This has to do with that a `NativeMessagingHosts` file isn't being created for Chromium, but does for those other browsers.
> > To fix this you can copy the `com.apple.passwordmanager.json` from Vivaldi/Brave's `NativeMessagingHosts` folder into Chromium's, or simply run the following command in Terminal:
> > echo '{
> > "name": "com.apple.passwordmanager",
> > "description": "PasswordManagerBrowserExtensionHelper",
> > "path": "/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper",
> > "type": "stdio",
> > "allowed_origins": [
> > "chrome-extension://pejdijmoenmkgeppbflobdenhhabjlaj/",
> > "chrome-extension://mfbcdcnpokpoajjciilocoachedjkima/"
> > ]
> > }' > ~/Library/Application\ Support/Chromium/NativeMessagingHosts/com.apple.passwordmanager.json

This origin is not required: `chrome-extension://mfbcdcnpokpoajjciilocoachedjkima/`

```json
{
  "name": "com.apple.passwordmanager",
  "description": "PasswordManagerBrowserExtensionHelper",
  "path": "/System/Cryptexes/App/System/Library/CoreServices/PasswordManagerBrowserExtensionHelper.app/Contents/MacOS/PasswordManagerBrowserExtensionHelper",
  "type": "stdio",
  "allowed_origins": ["chrome-extension://pejdijmoenmkgeppbflobdenhhabjlaj/"]
}
```
